### PR TITLE
Use external ERC821 contracts

### PR DESCRIPTION
**Pull request by mykelp**
https://github.com/hillstreetlabs/marketplace/issues/18

### CODE REVIEW

File: app/ERC821Contracts/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export {
```
Comment (pfletcherhill) [inline app/ERC821Contracts/index.js]: Like @mertcelebi had done before, can you make the folder structure `contracts/erc821/index.js`? Just leaving the option for adding `721` or others later.

File: app/store/index.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { action, observable } from "mobx";
 import contract from "truffle-contract";
 import AuctionBaseContract from "../../build/contracts/AuctionBase.json";
 import HillCoreContract from "../../build/contracts/HillCore.json";
+import { HILL_CORE_ABI, HILL_CORE_ADDRESS } from "../ERC821Contracts";
```
Comment (pfletcherhill) [inline app/store/index.js]: It'd be nicer to do `import { HillCore } from "../contracts/erc821"` and then be able to reference `HillCore.abi` and `HillCore.address`. Can you make that change?

File: app/components/Account.js [hunk 3]
```diff
@@ -11,53 +11,81 @@ export default class Account extends Component {
...
     window.s = this;
-    this.getCryptoHillsBalance();
-    const blockWatcher = observe(this.props.store, "currentBlock", change => {
-      this.getCryptoHillsBalance();
-    });
-    const balanceWatcher = observe(this, "cryptoHillsBalance", change => {
+    this.hillCoreWatcher = when(
+      () => this.props.store.hillCoreInstance,
+      () => {
+        this.hillCoreWatcher = null;
```
Comment (pfletcherhill) [inline app/components/Account.js]: What does this line do?

File: app/components/Account.js [hunk 4]
```diff
@@ -11,53 +11,81 @@ export default class Account extends Component {
...
+    );
+
+    this.balanceWatcher = observe(this, "cryptoHillsBalance", change => {
       this.getCryptoHills();
     });
   }
 
+  componentWillUnmount() {
+    if (this.hillCoreWatcher) {
+      this.hillCoreWatcher();
```
Comment (pfletcherhill) [inline app/components/Account.js]: Does just calling them like this cancel them?

File: app/store/index.js (hunk 2 above)
Comment (mykelp) [reply app/store/index.js]: Fixed

File: app/components/Account.js (hunk 4 above)
Comment (mykelp) [reply app/components/Account.js]: Yeah, `watch` and `autorun` return disposer functions that you can call in `componentWillUnmount`

File: app/components/Account.js (hunk 3 above)
Comment (mykelp) [reply app/components/Account.js]: Fixed

**Pull request merged**